neurosynth/analysis/reduce.py

```diff
@@ -155,3 +160,172 @@ def get_random_voxels(dataset, n_voxels):
+        n_words:    Number of top words to return for each topic. Default=31,
+                    based on Poldrack et al. (2012).
+        n_iters:    Number of iterations to run in training topic model.
+                    Default=1000.
+        alpha:      The Dirichlet prior on the per-document topic distributions.
+                    Default: 50 / n_topics, based on Poldrack et al. (2012).
+        beta:       The Dirichlet prior on the per-topic word distribution.
+                    Default: 0.001, based on Poldrack et al. (2012).
+    
+    Returns:
+        weights_df: A pandas DataFrame derived from the MALLET
+                    output-doc-topics output file. Contains the weight assigned
+                    to each article for each topic, which can be used to select
+                    articles for topic-based meta-analyses (accepted threshold
+                    from Poldrack article is 0.001). [n_topics]+1 columns:
+                    'pmid' is the first column and the following columns are
+                    the topic names. The names of the topics match the names
+                    in df (e.g., topic_000).
+        keys_df:    A pandas DataFrame derived from the MALLET
+                    output-topic-keys output file. Contains the top [n_words]
+                    words for each topic, which can act as a summary of the
+                    topic's content. Two columns: 'topic' and 'terms'. The
+                    names of the topics match the names in weights (e.g.,
+                    topic_000).
+    """
+    if abstracts.index.name != 'pmid':
+        abstracts.index = abstracts['pmid']
+
+    resdir = os.path.abspath(get_resource_path())
+    tempdir = os.path.join(resdir, 'topic_models')
+    absdir = os.path.join(tempdir, 'abstracts')
+    if not os.path.isdir(tempdir):
+        os.mkdir(tempdir)
+    
+    if alpha is None:
+        alpha = 50. / n_topics
+    
+    # Check for existence of MALLET and download if necessary
+    mallet_bin = os.path.join(resdir, 'mallet-2.0.7/bin/mallet')
+    if not os.path.isfile(mallet_bin):
+        print('MALLET toolbox not found. Downloading...')
+        cmd = ('curl -o {0}/mallet-2.0.7.tar.gz '
+               'http://mallet.cs.umass.edu/dist/mallet-2.0.7.tar.gz').format(resdir)
+        subprocess.call(cmd, shell=True)
+        cmd = 'cd {0}; tar zxf {0}/mallet-2.0.7.tar.gz'.format(resdir)
+        subprocess.call(cmd, shell=True)
+        os.remove(os.path.join(resdir, 'mallet-2.0.7.tar.gz'))
+    else:
+        print('MALLET toolbox found!')
+    
+    # Check for presence of abstract files and convert if necessary
+    if not os.path.isdir(absdir):
+        print('Abstracts folder not found. Creating abstract files...')
+        os.mkdir(absdir)
+        for pmid in abstracts.index.values:
+            abstract = abstracts.loc[pmid]['abstract']
+            with open(os.path.join(absdir, str(pmid)+'.txt'), 'wb') as fo:
```

Comment:
To be py35 compatible, should write as 'w' instead of 'wb.' Otherwise, will receive TypeError: a bytes-like object is required, not 'str'